### PR TITLE
feat(defaultOption): implements defaultOption

### PR DESCRIPTION
**Pull request by divyakumarjain**
- [X] Fix merge of boolean false value.
- [X] Fix merge of validation function
 - [ ] Make validation work
Closes #200

### CODE REVIEW

**Comment by divyakumarjain** [comment]
This is just an FYI...

If i dont get chance to work on this before I travel. You will have copy of work which is done so far.

File: src/core/services/formly.utils.ts [hunk 1]
```diff
@@ -19,8 +20,14 @@ export class FormlyUtils {
         return;
       }
       for (let srcArg in src) {
-        if (!dest[srcArg]) {
-          dest[srcArg] = this.clone(src[srcArg]);
```
Comment (divyakumarjain) [inline src/core/services/formly.utils.ts]: When  dest[srcArg] has false value it take value from source. Thus isLarge: false was replaced by isLarge:true from defaultTemplateOption

File: src/core/services/formly.utils.ts [hunk 2]
```diff
@@ -42,6 +57,6 @@ export class FormlyUtils {
     if (!this.isObject(value)) {
       return value;
     }
-    return Array.isArray(value) ? value.slice(0) : JSON.parse(JSON.stringify(value));
+    return Array.isArray(value) ? value.slice(0) : this.reverseDeepMerge({}, value);
   }
```
Comment (divyakumarjain) [inline src/core/services/formly.utils.ts]: JSON.parse(JSON.stringify(value) way of cloning does not copies function. Thus validation function were not cloned from defaultOptions. At present They get cloned in result of getMergedType but its not yet working on UI.

File: src/core/services/formly.utils.ts [hunk 3]
```diff
@@ -29,6 +36,14 @@ export class FormlyUtils {
     return dest;
   }
 
+  private isFunction(propertyValue: any) {
+    return typeof(propertyValue) === 'function'
+  }
+
+  private isBlankString(propertyValue: any) {
+    return propertyValue === "";
```
Comment (mohammedzamakhan) [inline src/core/services/formly.utils.ts]: Can you change this to single quotes

File: src/core/services/formly.utils.ts (hunk 2 above)
Comment (mohammedzamakhan) [inline src/core/services/formly.utils.ts]: Using reverseDeepMerge doesn't sound good to me. Can you try Object.assign or anything instead. Or have a check for a function?

File: src/core/services/formly.utils.ts (hunk 1 above)
Comment (mohammedzamakhan) [inline src/core/services/formly.utils.ts]: 👍

**Comment by mohammedzamakhan** [comment]
@divyakumarjain I applied your changes and updated the PR in the original repository. Thanks a lot 👍

**Pull request closed without merging**